capsules/core/src/test/capsule_test.rs

```diff
@@ -0,0 +1,54 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! Interface for running capsule tests.
+//!
+//! As Tock capsules are asynchronous, it is difficult for a test runner to
+//! determine when a test when a test has finished. This interface provides a
+//! `done()` callback to notify when the test is done.
```

Comment:
`done()` callback to be used by testing capsule to notify when the test is done.